src/Sylius/Behat/Context/Ui/Admin/ProductShowPageContext.php

```diff
@@ -0,0 +1,337 @@
+use Sylius\Behat\Element\Product\ShowPage\MoreDetailsElementInterface;
+use Sylius\Behat\Element\Product\ShowPage\OptionsElementInterface;
+use Sylius\Behat\Element\Product\ShowPage\PricingElementInterface;
+use Sylius\Behat\Element\Product\ShowPage\ShippingElementInterface;
+use Sylius\Behat\Element\Product\ShowPage\TaxonomyElementIterface;
+use Sylius\Behat\Element\Product\ShowPage\VariantsElementInterface;
+use Sylius\Behat\Page\Admin\Product\IndexPageInterface;
+use Sylius\Behat\Page\Admin\Product\ShowPageInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Webmozart\Assert\Assert;
+
+final class ProductShowPageContext implements Context
+{
+    /** @var IndexPageInterface */
+    private $indexPage;
+
+    /** @var ShowPageInterface */
+    private $productShowPage;
+
+    /** @var AssociationsElementInterface */
+    private $associationsElement;
+
+    /** @var AttributesElementInterface */
+    private $attributesElement;
+
+    /** @var DetailsElementInterface */
+    private $detailsElement;
+
+    /** @var MediaElementInterface */
+    private $mediaElement;
+
+    /** @var MoreDetailsElementInterface */
+    private $moreDetailsElement;
+
+    /** @var PricingElementInterface */
+    private $pricingElement;
+
+    /** @var ShippingElementInterface */
+    private $shippingElement;
+
+    /** @var TaxonomyElementIterface */
+    private $taxonomyElement;
+
+    /** @var OptionsElementInterface */
+    private $optionsElement;
+
+    /** @var VariantsElementInterface */
+     private $variantsElement;
+
+    public function __construct(
+        IndexPageInterface $indexPage,
+        ShowPageInterface $productShowPage,
+        AssociationsElementInterface $associationsElement,
+        AttributesElementInterface $attributesElement,
+        DetailsElementInterface $detailsElement,
+        MediaElementInterface $mediaElement,
+        MoreDetailsElementInterface $moreDetailsElement,
+        PricingElementInterface $pricingElement,
+        ShippingElementInterface $shippingElement,
+        TaxonomyElementIterface $taxonomyElement,
+        OptionsElementInterface $optionsElement,
+        VariantsElementInterface $variantsElement
+    ) {
+        $this->indexPage = $indexPage;
+        $this->productShowPage = $productShowPage;
+        $this->associationsElement = $associationsElement;
+        $this->attributesElement = $attributesElement;
+        $this->detailsElement = $detailsElement;
+        $this->mediaElement = $mediaElement;
+        $this->moreDetailsElement = $moreDetailsElement;
+        $this->pricingElement = $pricingElement;
+        $this->shippingElement = $shippingElement;
+        $this->taxonomyElement = $taxonomyElement;
+        $this->optionsElement = $optionsElement;
+        $this->variantsElement = $variantsElement;
+    }
+
+    /**
+     * @Given I am browsing products
+     */
+    public function iAmBrowsingProducts(): void
```

Comment:
This is already defined [there](https://github.com/Sylius/Sylius/blob/22b4e456faa972df8e17c1f0c7491125341658f8/src/Sylius/Behat/Context/Ui/Admin/ManagingProductsContext.php#L252)